qr-code/src/main/java/com/adyen/checkout/qrcode/FullQRCodeView.kt

```diff
@@ -0,0 +1,212 @@
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr), ComponentView {
+
+    private val binding: FullQrcodeViewBinding = FullQrcodeViewBinding.inflate(LayoutInflater.from(context), this)
+
+    private lateinit var localizedContext: Context
+
+    private val fileDownloader: FileDownloader = FileDownloader()
+
+    private lateinit var delegate: QRCodeDelegate
+
+    init {
+        orientation = VERTICAL
+        val padding = resources.getDimension(R.dimen.standard_double_margin).toInt()
+        setPadding(padding, padding, padding, padding)
+    }
+
+    override fun initView(delegate: ComponentDelegate, coroutineScope: CoroutineScope, localizedContext: Context) {
+        if (delegate !is QRCodeDelegate) throw IllegalStateException("Unsupported delegate type")
+
+        this.localizedContext = localizedContext
+        initLocalizedStrings(localizedContext)
+
+        this.delegate = delegate
+
+        observeDelegate(delegate, coroutineScope)
+
+        binding.buttonSaveImage.setOnClickListener {
+            val askedPermission = Manifest.permission.WRITE_EXTERNAL_STORAGE
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q &&
+                ContextCompat.checkSelfPermission(context, askedPermission) != PackageManager.PERMISSION_GRANTED
+            ) {
+                delegate.onError(
+                    PermissionException(
+                        errorMessage = "$askedPermission permission not granted",
+                        askedPermission = askedPermission
+                    )
+                )
+                return@setOnClickListener
+            }
+            saveQrImage(delegate.outputData.qrCodeData)
+        }
+    }
+
+    private fun initLocalizedStrings(localizedContext: Context) {
+        binding.buttonSaveImage.setLocalizedTextFromStyle(R.style.AdyenCheckout_QrCode_SaveButton, localizedContext)
+    }
+
+    private fun observeDelegate(delegate: QRCodeDelegate, coroutineScope: CoroutineScope) {
+        delegate.outputDataFlow
+            .onEach { outputDataChanged(it) }
+            .launchIn(coroutineScope)
+
+        delegate.timerFlow
+            .onEach { onTimerTick(it) }
+            .launchIn(coroutineScope)
+    }
+
+    private fun outputDataChanged(outputData: QRCodeOutputData) {
+        Logger.d(TAG, "outputDataChanged")
+
+        updateMessageText(outputData.paymentMethodType)
+        updateLogo(outputData.paymentMethodType)
+        updateQrImage(outputData.qrCodeData)
+        updateAmount(delegate.componentParams)
+    }
+
+    private fun updateAmount(componentParams: ComponentParams) {
+        if (!componentParams.amount.isEmpty) {
+            val formattedAmount = CurrencyUtils.formatAmount(
+                componentParams.amount,
+                componentParams.shopperLocale
+            )
+            binding.textviewAmount.visibility = View.VISIBLE
```

Comment:
Should use `isVisible = true`

Reply:
what's the difference?

Reply:
Just for consistency, we use `isVisible` everywhere